providers/file/provider.go

```diff
@@ -232,7 +236,11 @@ func (p *Provider) removeCluster(name string) {
 	}
 }
 
-func (p *Provider) run(ctx context.Context, mgr mcmanager.Manager) error {
+func (p *Provider) run(ctx context.Context) error {
+	if p.aware == nil {
+		return fmt.Errorf("aware is not set")
+	}
+
```

Comment:
```suggestion
```

The file provider is actually intended to be able to run without something that can engage the cluster